ui/pages/asset/components/asset-v2.tsx

```diff
@@ -0,0 +1,423 @@
+import { MINUTE } from '../../../../shared/constants/time';
+import TokenCell from '../../../components/app/token-cell';
+import TransactionList from '../../../components/app/transaction-list';
+import Tooltip from '../../../components/ui/tooltip';
+import AssetChart from './asset-chart';
+import { getPricePrecision, localizeLargeNumber } from './util';
+import AssetHeader from './asset-header';
+
+const renderRow = (leftColumn: string, rightColumn: ReactNode) => (
+  <Box display={Display.Flex} justifyContent={JustifyContent.spaceBetween}>
+    <Text color={TextColor.textAlternative} variant={TextVariant.bodyMdMedium}>
+      {leftColumn}
+    </Text>
+    {rightColumn}
+  </Box>
+);
+
+const renderTooltip = (button: JSX.Element, text: string, show: boolean) =>
+  show ? (
+    <Tooltip wrapperClassName="asset-tooltip" title={text} position="bottom">
+      {button}
+    </Tooltip>
+  ) : (
+    <Box width={BlockSize.Full}>{button}</Box>
+  );
+
+// A page representing details for a native or token asset
+const AssetV2 = ({
+  asset,
+}: {
+  asset: (
+    | {
+        type: AssetType.native;
+        isOriginalNativeSymbol: boolean;
+      }
+    | {
+        type: AssetType.token;
+        address: string;
+        decimals: number;
+        aggregators?: [];
+      }
+  ) & {
+    symbol: string;
+    name?: string;
+    image: string;
+    balance?: string;
+    fiatDisplay?: string;
+    optionsButton: React.ReactNode;
+  };
+}) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+  const chainId = hexToDecimal(useSelector(getCurrentChainId));
+  const currency = useSelector(getCurrentCurrency);
+  const isSwapsChain = useSelector(getIsSwapsChain);
+  const isBridgeChain = useSelector(getIsBridgeChain);
+  const isBuyableChain = useSelector(getIsBuyableChain);
+
+  const [marketData, setMarketData] = useState<any>();
+  const [balanceInView, setBalanceInView] = useState(true);
+
+  const balanceRef = useRef(null);
+
+  const { type, symbol, name, image, balance, fiatDisplay, optionsButton } =
+    asset;
+
+  const address =
+    type === AssetType.token
+      ? asset.address
+      : '0x0000000000000000000000000000000000000000';
+
+  useEffect(() => {
+    setMarketData(undefined);
+
+    // TODO: Consider exposing HTTP request through a controller
+    fetchWithCache({
+      url: `https://price-api.metafi.codefi.network/v2/chains/${chainId}/spot-prices/?includeMarketData=true&tokenAddresses=${address}&vsCurrency=${currency}`,
+      cacheOptions: { cacheRefreshTime: MINUTE },
+      functionName: 'GetAssetMarketData',
+    })
+      .catch(() => null)
+      .then((data) => setMarketData(data?.[address.toLowerCase()]));
+  }, [chainId, address, currency]);
+
+  // Show balance in header when it's no longer in view
+  useEffect(() => {
```

Comment:
Design feedback: Have the balance fade in/out instead of just on/off.  Might have to do a scroll listener instead of this observer approach, since the observer is just called once as it goes in/out of view instead of knowing the %.